### ISSUE TITLE
Decal ORM implementation not completely overriding source texture?

### ISSUE THREAD

**Issue by everythingandnothingdev** [godotengine/godot#51299]
### Godot version

4.0dev (1f69582)

### System information

Windows 10, Vulkan, RTX 2080

### Issue description

I'm trying to create a "puddle" decal that's painted on top of a street. A common case in video games, the puddle needs to be a pure mirror, while the street, not. 

The street roughness appears to affect the decal roughness, so I can't achieve this effect.

### Steps to reproduce

I created a simple ORM texture with no ambient occlusion, full gloss, full metallic, basically a mirror.

![image](https://user-images.githubusercontent.com/87843097/128435454-d0ee777b-356b-4af2-b63b-efda4c016f8f.png)

The resulting ORM texture is magenta.

![image](https://user-images.githubusercontent.com/87843097/128435479-a0c89349-73cf-48de-942c-cd86084a865c.png)

Now I've placed a puddle decal on the street that has this ORM texture, when the street itself has "1" metallic and "0" roughness, I see a reflection across the entire street, including the puddle decal.

(Street texture settings)
![image](https://user-images.githubusercontent.com/87843097/128435626-b07f9e6a-b8c1-4b64-899d-146f02dd457c.png)

![image](https://user-images.githubusercontent.com/87843097/128435581-3685092f-df71-400a-a5f9-948946e15b3c.png)

However, when I turn the roughness up on the street texture, it also affects the puddle texture to the point the puddle is no longer reflective.

(Street texture settings)
![image](https://user-images.githubusercontent.com/87843097/128435885-62f28f8d-9bdd-4a4f-8956-3cd052905ebc.png)

![image](https://user-images.githubusercontent.com/87843097/128435841-925065e1-7f40-41ce-bf55-f742fb2e4422.png)

I don't know what the logic is behind how much a source texture affects the decal, but I basically need control over how this blends the ORM texture.

### Minimal reproduction project

[Decal ORM Bug.zip](https://github.com/godotengine/godot/files/6943335/Decal.ORM.Bug.zip)

Left: white plane has roughness: 1
Right: white plane has roughness: 0
Expected: both black decals show a reflection of the orange cube

![image](https://user-images.githubusercontent.com/87843097/128463075-02e74b5d-3c65-4c09-937f-4524a302fd84.png)

**Comment by Calinou** [comment]
It works in the [GodotCon 2021 decals demo](https://github.com/Calinou/godotcon-2021-decals) I created. Check the splat texture I used.

**Comment by everythingandnothingdev** [comment]
Try my reproduction on master and let me know if I did something wrong. The decal doesn't display in the first place if I have no albedo texture.

**Comment by everythingandnothingdev** [comment]
@Calinou 

I looked into this a little bit. For one thing, your GodotCon 2021 decals demo does not have SDFGI or screen space reflections enabled to be able to see the reflections in the blood splatters, so that does not apply here.

To the best of my understanding, the roughness value from the ORM texture never seems to make it to the SDFGI calculation of the reflected light.

https://github.com/godotengine/godot/blob/2d446771d6701bb77432b6df5a768e53c9a6c95d/servers/rendering/renderer_rd/shaders/gi.glsl#L593

If I replace "roughness" in the line above with "0.0", I see the reflection on both black decals in my reproduction. Tracing back up, that roughness value is computed based solely on this "normal_roughness_buffer" variable. Which I can only assume is the normal texture for the current material, since I can't find a case where it is modified in any other part of the code, aside from allocation.

https://github.com/godotengine/godot/blob/2d446771d6701bb77432b6df5a768e53c9a6c95d/servers/rendering/renderer_rd/shaders/gi.glsl#L574

So basically this `sdfgi_process()` function needs to be passed a roughness value that is modified based on the ORM texture's roughness.

**Comment by clayjohn** [comment]
So I looked into this a little bit. ``process_gi`` is only used when we are using "deferred GI" which means that the GI is computed in a full screen (half res) pass before rendering objects. Deferred GI doesn't have access to the decal buffer because it runs in screen space and uses the normal_roughness buffer. 

Accordingly, I'm not certain this can be fixed, with a little bit more searching, I think we can find a workaround though (e.g. forcing GI to be computed in the forward pass)

**Comment by HarDevSan** [comment]
But even with the Albedo Blending it is weird, as lowering the colors modulate property value will reveal the underlying surface as well, despite the Albede Mix being at full opacity of 1.0:

https://www.dropbox.com/scl/fi/wlqnma2h12gkug2kv5q0k/2026-03-03-16-15-48.mp4?rlkey=i5rjffg87wijvqws6jnjuwu7s&dl=0

That also seems not correct to me, because this would mean that stickers and such, when modulated, become see-through.

It is even worse than that, cause if you use a non white Albedo texture inside a decal in Godot, the underlying surface comes through as well, even tho the Modulate property is full white and Albedo Mix at 1: 

<img width="2560" height="1369" alt="Image" src="https://github.com/user-attachments/assets/a9ef4a33-2c1d-46d2-a87d-82443ed90339" />

**Comment by Glorax** [comment]
Why has this been closed as completed? Just because the limitation has finally been documented doesn't mean the underlying issue is solved.

What's more, as @HarDevSan says above, the claim in the documentation that the ORM texture functions with Reflection Probes may not even be true.

**Comment by Glorax** [comment]
> Also, yes, as @Glorax said, please have a look at the foundational issue with color and alpha in the current decal implementation, which I demonstrated above: https://github.com/godotengine/godot/issues/51299#issuecomment-3991742470 (+dropbox link to video)

You have turned the upper and lower vertical fade to 0.0, correct?

By default they're both set to 0.3, which results in decals being partially transparent.
